### PR TITLE
ref(tests): Restructure and consolidate event creation in tests

### PR DESCRIPTION
**Pull request by tkaemming**
This also nudges `EventsProcessorBase` in the direction of being more accurately typed.

### CODE REVIEW

File: snuba/datasets/events_processor_base.py [hunk 1]
```diff
@@ -29,26 +31,42 @@
 logger = logging.getLogger(__name__)
 
 
+EventData = Mapping[str, Any]
+
+
+class InsertEvent(TypedDict):
```
Comment (tkaemming) [inline snuba/datasets/events_processor_base.py:37]: This is named as such just to delineate it from replacement events that come through the same consumer.

Comment (tkaemming) [reply snuba/datasets/events_processor_base.py:37]: Also worth noting this is reverse engineered from [`SnubaProtocolEventStream`](https://github.com/getsentry/sentry/blob/cb3a13232991550fd83d584630ede6529046c555/src/sentry/eventstream/snuba.py#L110-L124), so it does not include `search_message` which is referenced below and causes a mypy error.

File: tests/base.py [hunk 2]
```diff
@@ -97,71 +68,90 @@ def teardown_method(self, test_method):
...
+    def write_events(self, events: Sequence[Union[EventData, InsertEvent]]) -> None:
+        processor = (
+            enforce_table_writer(self.dataset).get_stream_loader().get_processor()
+        )
 
-        rows = []
+        processed_messages = []
         for event in events:
-            rows.append(event)
+            if self.__is_event_data(event):
```
Comment (fpacifici) [inline tests/base.py:147]: Is there actually a legit case where this may return false and the test should not fail ?
Can't we assert this?

**Review by fpacifici — CHANGES_REQUESTED** [review]
I think there is a mistake on the type of group_id in `InsertedEvent` otherwise looks good.

File: snuba/datasets/events_processor_base.py [hunk 3]
```diff
@@ -29,26 +31,42 @@
...
+    group_id: int
```
[the remark this answers is not in the document]
Comment (tkaemming) [reply snuba/datasets/events_processor_base.py:53]: Ah, good point. Is that also true for `primary_hash`?

File: tests/base.py (hunk 2 above)
Comment (tkaemming) [reply tests/base.py:147]: I thought there still was, but I guess I actually changed all of the inputs to this function to `InsertEvent` and didn't even realize it. Cool, I'll remove this conditional. (I also felt like `__is_event_data` was a super hacky function so it's nice to be able to remove it as well.)

Comment (tkaemming) [reply tests/base.py]: Cleaned up here: b23b209a608619f5d7481deac60c77d270deb7af

It's not very easy to assert on whether or something satisfies a `TypedDict` at runtime so going to punt on that for now, I'll likely be changing this into a different data structure soon enough as well (foreshadowing.…)

File: snuba/datasets/events_processor_base.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (fpacifici) [reply snuba/datasets/events_processor_base.py:53]: I don't think so. primary_hash seems always populated, whether intentionally or not
`primary_hash=job["data"]["hashes"][0] if "hashes" in job["data"] else "",`

[the remark this answers is not in the document]
Comment (tkaemming) [reply snuba/datasets/events_processor_base.py]: Fixed with 5d32694 and also noted the `primary_hash` semantics in a comment and created a task to address it later.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**